src/Integration.UnitTests/Connection/ConnectionWorkflowTests.cs

```diff
@@ -86,11 +86,20 @@ public void TestInit()
 
         #region Tests
 
+        [TestMethod]
+        public void ConnectionWorkflow_ConnectionStep_WhenGivenANullServiceProvider_ThrowsArgumentNullException()
```

Comment:
New low-value test, but I added it since there were already tests for the other constructor arguments.